src/components/ChangeDataForm.vue

```diff
@@ -0,0 +1,262 @@
+							<ul class="pl-2 ml-2 mt-0" style="line-height: 1.5">
+								<li>At least one lowercase</li>
+								<li>At least one uppercase</li>
+								<li>At least one numeric</li>
+								<li>Minimum 8 characters</li>
+							</ul>
+						</template>
+					</Password>
+				</template>
+			</Inplace>
+			<Inplace :closable="true" class="inplace">
+				<template #display>
+					<span class="label">{{ patientData.address.street }}</span>
+					<span class="pi pi-pencil" style="vertical-align: middle"></span>
+				</template>
+				<template #content>
+					<InputText v-model="patientData.address.street" autoFocus />
+				</template>
+			</Inplace>
+			<Inplace :closable="true" class="inplace">
+				<template #display>
+					<span class="label">{{ patientData.address.houseNumber }}</span>
+					<span class="pi pi-pencil" style="vertical-align: middle"></span>
+				</template>
+				<template #content>
+					<InputText v-model="patientData.address.houseNumber" autoFocus />
+				</template>
+			</Inplace>
+			<Inplace :closable="true" class="inplace">
+				<template #display>
+					<span class="label">{{ patientData.address.localNumber }}</span>
+					<span class="pi pi-pencil" style="vertical-align: middle"></span>
+				</template>
+				<template #content>
+					<InputText v-model="patientData.address.localNumber" autoFocus />
+				</template>
+			</Inplace>
+			<Inplace :closable="true" class="inplace">
+				<template #display>
+					<span class="label">{{ patientData.address.city }}</span>
+					<span class="pi pi-pencil" style="vertical-align: middle"></span>
+				</template>
+				<template #content>
+					<InputText v-model="patientData.address.city" autoFocus />
+				</template>
+			</Inplace>
+			<Inplace :closable="true" class="inplace">
+				<template #display>
+					<span class="label">{{ patientData.address.zipCode }}</span>
+					<span class="pi pi-pencil" style="vertical-align: middle"></span>
+				</template>
+				<template #content>
+					<InputMask v-model="patientData.address.zipCode" mask="99-999" />
+				</template>
+			</Inplace>
+			<Button class="" label="Confirm" @click="compareData" :loading="isLoading" />
+		</template>
+	</Card>
+</template>
+
+<script setup>
+import { useUserSession } from "@/services/useUserSession";
+import Inplace from "primevue/inplace";
+import InputText from "primevue/inputtext";
+import Card from "primevue/card";
+import InputMask from "primevue/inputmask";
+import Button from "primevue/button";
+import { changeData } from "@/services/api";
+import { useToast } from "primevue/usetoast";
+import Password from "primevue/password";
+import Divider from "primevue/divider";
+import Cookies from "js-cookie";
+import { ref, defineProps, defineEmits } from "vue";
+
+defineProps({
+	visible: {
+		type: Boolean,
+		default: false,
+	},
+});
+const emit = defineEmits(["hide"]);
+
+const isLoading = ref(false);
+const toast = useToast();
+
+const compare = (obj1, obj2, properties = {}, objKey = null) => {
```

Comment:
Można to przepisać żeby było bardziej czytelne. Taka modyfikacja argumentów to zła praktyka (bo w ogólnym przypadku nie mamy gwarancji że dostaniemy skopiowany obiekt, a raczej nie chcemy żeby efektem ubocznym naszej funkcji była zmiana stanu)
Zobacz wersję z ostatniego commita